### PR TITLE
use heap.Interface with heap.Pop()

### PR DESCRIPTION
**Pull request by mixmasala**
depends on https://github.com/katzenpost/core/pull/74

### CODE REVIEW

File: internal/scheduler/queue_mem.go [hunk 1]
```diff
@@ -50,7 +51,7 @@ func (q *memoryQueue) Peek() (time.Duration, *packet.Packet) {
 }
 
 func (q *memoryQueue) Pop() {
-	q.q.Pop()
+	heap.Pop(q.q)
```
Comment (david415) [inline internal/scheduler/queue_mem.go]: is this better?

Comment (mixmasala) [reply internal/scheduler/queue_mem.go]: yes, from the container/heap documentation:
```
// Note that Push and Pop in this interface are for package heap's
// implementation to call. To add and remove things from the heap,
// use heap.Push and heap.Pop.
```
and the reason:
```
func Pop(h Interface) interface{} {
        n := h.Len() - 1
        h.Swap(0, n)
        down(h, 0, n)
        return h.Pop()
}
```

Comment (mixmasala) [reply internal/scheduler/queue_mem.go]: I have a suspicion that the Peek() method isn't correct, but I'm not sure about it.

Comment (david415) [reply internal/scheduler/queue_mem.go]: write a test?

**Comment by mixmasala** [comment]
https://github.com/katzenpost/server/pull/113 adds a unit test for the in memory queue that uses the PriorityQueue

**Pull request merged**